consensus/core/src/dag_state.rs

```diff
@@ -329,6 +438,29 @@ impl DagState {
         }
     }
 
+    /// Detects and returns the blocks of the round that forms the last quorum. The method will return
+    /// the quorum even if that's genesis.
+    pub(crate) fn last_quorum(&self) -> Vec<VerifiedBlock> {
+        // the quorum should exist either on the highest accepted round or the one before. If we fail to detect
+        // a quorum then it means that our DAG has advanced with missing causal history.
+        for round in
+            (self.highest_accepted_round.saturating_sub(1)..=self.highest_accepted_round).rev()
+        {
+            if round == GENESIS_ROUND {
+                return self.genesis_blocks();
+            }
+            let mut quorum = StakeAggregator::<QuorumThreshold>::new();
+            let blocks = self.get_uncommitted_blocks_at_round(round);
```

Comment:
Worth adding a comment that round must be uncommitted because minimum wave length is 3.

Reply:
Sure can add this